llvm/tools/append-file/append-file.cpp

```diff
@@ -52,17 +63,45 @@ int main(int argc, const char **argv) {
   if (!llvm::sys::fs::exists(Input))
     error("input file not found");
 
-  // Copy the input file to the output file
-  llvm::sys::fs::copy_file(Input, Output);
+  // Open the original source file stream.
+  std::ifstream InputFile(Input, std::ios_base::binary);
+  // Open the output file stream.
+  std::ofstream OutFile(Output, std::ios_base::binary);
+
+  // Add the BOM from the original source (if it exists).  We will handle
+  // UTF-8, UTF-16LE and UTF-16BE
+  unsigned char char1 = InputFile.get();
```

Comment:
(Minor) suggestion of the guideline-conforming renaming
```suggestion
  unsigned char Char1 = InputFile.get();
```